federated_content_connector/filters/pipeline.py

```diff
@@ -76,3 +79,46 @@ def run_filter(self, course_key, serialized_enrollment):  # pylint: disable=argu
             pass
 
         return {'course_key': course_key, 'serialized_enrollment': serialized_enrollment}
+
+
+class CreateApiRenderCourseRunStep(PipelineStep):
+    """
+    Step that modifies the courserun data for the course.
+
+    Example usage:
+
+    Add the following configurations to your configuration file:
+
+        "OPEN_EDX_FILTERS_CONFIG": {
+            "org.openedx.learning.home.courserun.api.rendered.started.v1": {
+                "fail_silently": False,
+                "pipeline": [
+                    "federated_content_connector.filters.pipeline.CreateApiRenderCourseRunStep"
+                ]
+            }
+        }
+    """
+
+    def run_filter(self, serialized_courserun):  # pylint: disable=arguments-differ
+        """
+        Pipeline step that modifies the courserun data for the course.
+        """
+        try:
+            course_details = CourseDetails.objects.get(id=serialized_courserun.get('courseId'))
+            course_type = course_details.course_type
+            product_source = course_details.product_source
+            start_date, end_date = course_details.start_date, course_details.end_date
+
+            if product_source == PRODUCT_SOURCE_2U and course_type == EXEC_ED_COURSE_TYPE:
+                now_utc = datetime.now(utc)
+                serialized_courserun.update({
+                    'startDate': start_date,
+                    'endDate': end_date,
+                    'isStarted': now_utc > start_date if start_date is not None else True,
+                    'isArchived': now_utc > end_date if end_date is not None else False
```

Comment:
As discussed on call, we need to verify in which cases start_date and end_date can be null and what is correct status in that case?

Reply:
I had a discussion with the Pheonix member and I found that start_date will not be null for any course run. But as we see in our CourseDetail model there are some EE course_runs which has null start_date. So until this issue has been resolved in CourseDetail I would like to keep this extra check.